crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1832,23 +1840,55 @@ impl<'db> ClassType<'db> {
     /// Note that any class (except for `object`) that has no explicit
     /// bases will implicitly inherit from `object` at runtime. Nonetheless,
     /// this method does *not* include `object` in the bases it iterates over.
-    fn explicit_bases(self, db: &'db dyn Db) -> impl Iterator<Item = Type<'db>> {
-        let definition = self.definition(db);
+    ///
+    /// ## Why is this a salsa query?
+    ///
+    /// This is a salsa query to short-circuit the invalidation
+    /// when the class's AST node changes.
+    ///
+    /// Were this not a salsa query, then the calling query
+    /// would depend on the class's AST and rerun for every change in that file.
+    fn explicit_bases(self, db: &'db dyn Db) -> &[Type<'db>] {
+        self.explicit_bases_query(db)
+    }
+
+    #[salsa::tracked(return_ref)]
+    fn explicit_bases_query(self, db: &'db dyn Db) -> Vec<Type<'db>> {
```

Comment:
nit: worth returning `Box<[Type<'db>]>` here rather than `Vec<Type<'db>>`? It might result in slightly less data being stored in the Salsa cache?